projects/core/src/user/facade/user-consent.service.ts

```diff
@@ -236,12 +236,13 @@ export class UserConsentService {
    * Withdraw consent for the given `consentCode`
    * @param consentCode for which to withdraw the consent
    */
-  withdrawConsent(consentCode: string): void {
+  withdrawConsent(consentCode: string, consentId?: string): void {
     this.userIdService.takeUserId().subscribe((userId) => {
       this.store.dispatch(
         new UserActions.WithdrawUserConsent({
           userId,
           consentCode,
+          consentId,
```

Comment:
[importance: high]

Is the concept of `consentId` relevant for CX Commerce? Or it's only relevant for CDC?

a) If it's common to CDC and CX Commerce, I'm fine with adding everywhere this new optional parameter. 

b) If only to CDC, it looks strange we leak this information in the core service (facade), ngrx action, connector and adapter.
If the `consentId` is relevant only to CDC (and not CX Commerce), what do you think about just adding a generic argument `params?: Record<string,any>` as an extensibility point (instead of concrete `consentId:string`) in the core clases/methods?
And in your custom CDC-flavoured classes (that extend from the original ones), we would pass those params along the layers of service, ngrx, connector and adapter?

Reply:
consentId is applicable for CX commerce as well (field id in interface ConsentTemplate  in projects/core/src/model/consent.model.ts)

Reply:
did you mean 'relevant' as 'if it is being used in the functionality' or as 'if this property is present in core model' ?

Reply:
As far as I understand, consent ID is not used in the CX Commerce functionality, but it's present in the core model.
If that's true, I think it doesn't look that bad, if we add the ID parameter to the core services, because the ID is already present in the core model.

So let's resolve this comment.

Reply:
okay